### PR TITLE
Move PowerUtils to dedicated power management service

### PR DESCRIPTION
**Pull request by ezaquarii**
Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

File: src/main/java/com/nextcloud/client/device/Package.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Package com.nextcloud.client.device
```
Comment (ezaquarii) [inline src/main/java/com/nextcloud/client/device/Package.md]: This is Kotlin (KDoc to be specific) way of docimenting packages and modules.

File: src/main/java/com/nextcloud/client/device/DeviceInfo.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.nextcloud.client.device
+
+import android.os.Build
+
+class DeviceInfo {
```
Comment (ezaquarii) [inline src/main/java/com/nextcloud/client/device/DeviceInfo.kt]: This should save us from rolling some `PowerMock` kludge test API-level-dependent code.

Nevertheless there will be a point in time where mocking platform code will require `Robolectric`, but I'd rather delay that until it's really needed.

File: src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 3]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (ezaquarii) [inline src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Classes are `final` in Kotlin by default, so mocking require specialized mock maker.

**Comment by AndyScherzinger** [comment]
@ezaquarii nice work, looking good to me. Can you add the license header to files? :+1;

**Comment by tobiasKaminsky** [comment]
@ezaquarii thanks for the detailed explanation on new kotlin stuff :-)

I added gpl license header, so that I can merge it once CI is green.

**Pull request merged**